packages/blade/src/components/DatePicker/Calendar.web.tsx

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import dayjs from 'dayjs';
+import React from 'react';
+import type { CalendarLevel } from '@mantine/dates';
+import { shiftTimezone, useDatesContext, DatePicker } from '@mantine/dates';
+import type { CalendarProps, DateSelectionType, PickerType } from './types';
+import { CalendarHeader } from './CalendarHeader';
+import { CalendarGradientStyles, CalendarStyles } from './CalendarStyles';
+import { useUncontrolledDates } from './useControlledDates';
+import { levelToPicker, pickerToLevel, classes } from './constants';
+import { useControllableState } from '~utils/useControllable';
+import { useIsMobile } from '~utils/useIsMobile';
+import { throwBladeError } from '~utils/logger';
+
+const Calendar = <Type extends DateSelectionType>({
+  firstDayOfWeek = 0,
+  selectionType,
+  allowSingleDateInRange,
+  defaultPicker = 'day',
+  picker,
+  onPickerChange,
+  date,
+  defaultDate,
+  onDateChange,
+  locale,
+  onNext,
+  onPrevious,
+  onNextMonth,
+  onNextYear,
+  onNextDecade,
+  onPreviousMonth,
+  onPreviousYear,
+  onPreviousDecade,
+  presets,
+  ...props
+}: CalendarProps<Type> & {
+  date?: Date;
+  defaultDate?: Date;
+  onDateChange?: (date: Date) => void;
+}): React.ReactElement => {
+  const isRange = selectionType === 'range';
+
+  const [level, setLevel] = useControllableState<CalendarLevel>({
+    defaultValue: pickerToLevel[defaultPicker],
+    value: pickerToLevel[picker!],
+    onChange: (level) => {
+      onPickerChange?.(levelToPicker[level]);
+    },
+  });
+
+  if (__DEV__) {
+    if (isRange && level !== 'month') {
+      throwBladeError({
+        message: `Cannot use range DatePicker with pickerType: ${levelToPicker[level]}, Only "day" is supported`,
+        moduleName: 'DatePicker',
+      });
+    }
+  }
+
+  const [_date, setDate] = useUncontrolledDates({
+    type: 'default',
+    value: date,
+    defaultValue: defaultDate,
+    onChange: onDateChange as any,
+  });
+
+  const ctx = useDatesContext();
```

Comment:
User a better variable name?
```suggestion
  const dateContext = useDatesContext();
```

Reply:
done.